src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -275,7 +283,20 @@ private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string
                         apiResponseType.ApiResponseFormats.Add(defaultResponseFormat);
                     }
 
-                    supportedResponseTypes.Add(apiResponseType);
+                    bool foundDuplicateStatusCode = false;
+                    foreach (var existingResponseType in supportedResponseTypes)
+                    {
+                        if (existingResponseType.StatusCode.Equals(apiResponseType.StatusCode))
```

Comment:
Isn't this an `int`
```suggestion
                        if (existingResponseType.StatusCode == apiResponseType.StatusCode)
```

Reply:
I would use `Any` here rather than looping. This code isn't super performant to start with, so the allocation isn't something to worry about.,

```
var foundDuplicateStatusCode  = supportedResponseTypes.Any(existingResponseType => existingResponseType.StatusCode == apiResponseType.StatusCode);
```